Gems/ROS2/Code/Source/Gripper/FingerGripperComponent.cpp

```diff
@@ -0,0 +1,288 @@
+                        "Stall Time",
+                        "The time to wait before considering the gripper as stalled.");
+            }
+        }
+    }
+
+    AZStd::string GetJointName(AZ::EntityId entityId)
+    {
+        AZ::Entity* entity{ nullptr };
+        AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+        AZ_Assert(entity, "Entity %s not found.", entityId.ToString().c_str());
+        if (entity)
+        {
+            ROS2FrameComponent* component = entity->FindComponent<ROS2FrameComponent>();
+            if (component)
+            {
+                return component->GetJointName().GetStringView();
+            }
+        }
+        return AZStd::string();
+    }
+
+    ManipulationJoints& FingerGripperComponent::GetFingerJoints()
+    {
+        m_fingerJoints.clear();
+        m_rootOfArticulation = Utils::GetRootOfArticulation(GetEntityId());
+        AZ_Warning(
+            "FingerGripperComponent",
+            m_rootOfArticulation.IsValid(),
+            "Entity %s is not part of an articulation.",
+            GetEntity()->GetName().c_str());
+        ManipulationJoints allJoints;
+        if (m_rootOfArticulation.IsValid())
+        {
+            JointsManipulationRequestBus::EventResult(allJoints, m_rootOfArticulation, &JointsManipulationRequests::GetJoints);
+        }
+        AZStd::vector<AZ::EntityId> descendantIds;
+        AZ::TransformBus::EventResult(descendantIds, GetEntityId(), &AZ::TransformBus::Events::GetAllDescendants);
+
+        for (AZ::EntityId descendant : descendantIds)
+        {
+            AZStd::string jointName = GetJointName(descendant);
+            if (!jointName.empty())
+            {
+                AZ_Printf("FingerGripperComponent", "Adding finger joint %s", jointName.c_str());
+                m_fingerJoints[jointName] = allJoints[jointName];
+            }
+        }
+
+        return m_fingerJoints;
+    }
+
+    void FingerGripperComponent::SetPosition(float position, float maxEffort)
+    {
+        // A check to make this work with moveit
+        if (maxEffort == 0.0f)
+        {
+            maxEffort = AZStd::numeric_limits<float>::infinity();
+        }
+
+        float targetPosition = position;
+        for (auto& [jointName, jointInfo] : m_fingerJoints)
+        {
+            AZ::Outcome<void, AZStd::string> result;
+            JointsManipulationRequestBus::EventResult(
+                result, m_rootOfArticulation, &JointsManipulationRequests::MoveJointToPosition, jointName, targetPosition);
+            if (!result.IsSuccess())
+            {
+                AZ_Warning(
+                    "FingerGripperComponent",
+                    result,
+                    "Joint move cannot be realized: %s for %s ",
+                    result.GetError().c_str(),
+                    jointName.c_str());
+            }
+        }
+
+        float oneMaxEffort = maxEffort / m_fingerJoints.size();
```

Comment:
Please handle the case where m_fingerJoints is empty to avoid division by zero. This could be done by an early return at the start.